### ISSUE TITLE
Conditional negative lookahead inside positive lookahead fails to match

### ISSUE THREAD

**Issue by DexterHill0** [mrabarnett/mrab-regex#498]
I ran into an issue with a regex I was building which was causing the regex to return `None` when I would've expected a match. I narrowed it down to it possibly being an issue with having a conditional negative lookahead inside of a non-conditional positive lookahead.
A basic example of such a regex is:
```regex
(?=(?:foo)?(?(?!bar)(\w*)|()))
```
The two test strings are: `foobar` and `fooq`. Due to the empty capture group, in *both* cases there should be a match. In the case of `fooq`, as the negative lookeahead finds no `bar`, it will match `q`. `foobar` will match nothing but an empty string. 
It seems that in `regex`, it fails to match `fooq` and instead returns `None`. It does correctly match an empty string for `foobar`, however.
```py
test = regex.compile(r"(?=(?:foo)?(?(?!bar)(\w*)|()))")
print(test.match("fooq"))
# > None
print(test.match("foobar"))
# > <regex.Match object; span=(0, 0), match=''>
```

**Comment by mrabarnett** [comment]
I've simplified the issue to:

```
assert regex.match(r"(?(?=a).|..)", "ab").span() == (0, 1)
assert regex.match(r"(?(?=b).|..)", "ab").span() == (0, 2)
assert regex.match(r"(?(?!a).|..)", "ab").span() == (0, 2)
assert regex.match(r"(?(?!b).|..)", "ab").span() == (0, 1) # FAILS

```

**Comment by mrabarnett** [comment]
Fixed in regex 2023.6.3.

**Comment by DexterHill0** [comment]
awesome thanks!
